### ISSUE TITLE
Load first image and then rest in background

### ISSUE THREAD

**Issue by BorysTyminski** [deulos/vue-flux#8]
What do you think about idea to load first image of slider, hide loader gif and then load rest images in "background"? It could save some time to show first screen to user which is good user experience.

**Comment by BorysTyminski** [comment]
>  one is that the final loading time will be increased by first image load time 

I don't really see the reason why overall loading time (of entire slider) would rise by showing first image as soon as it's loaded. Maybe I'm wrong but you need to give more details why it would increase.

> other is that may give the false feeling of loaded and not working.

Maybe you're right especially when first image will load and user would need to wait for example 10-20 seconds to see next one. But think about that without showing first image he will wait same 10-20 seconds looking and blank page with spinner.

> There is way to achieve the same, setting the first image as background. But I really think the loader should stay until all images loaded, 

If I set first image as background of slider div by myself it would cause that I'm loading same image two times one in CSS as background and one in vue-flux slider by your code.

> Maybe I can show the % of images loaded to improve the user experience if you are worried about waiting time.

I don't know if it's possible to do due to loading image is not async backend call or something like that.

<hr>

I was thinking much more about it and I think best soloution would be to enable devs to set optional flag which would do that slider would work like this:

1. Set first image as background (not by myself but by your component).
2. Spinner still working.
3. Load other images.
4. Hide spinner and start slider "sliding".

Basically that's same as first idea but without hiding spinner so user will not have false feeling that "hey something not working fine!" because he will see that images are still loading.

And this behaviour would be optional so default behaviour would be same as it's now and second one would be above.

<hr>

Except that giving slot to enable devs to overwrite spinner is **must have** as blue colour spinner doesn't fit my page for example and much more pages.

**Comment by deulos** [comment]
I have created the slot spinner although if the only problem is the color you should just overwrite the CSS.

I also saw doing some tests that is nice to display the first image while the rest are loaded, so I did it that way. I don't imagine somebody would prefer to watch blank instead an image.

The spinner will disappear when all loaded.

You can check it in the documentations and if there are not more changes I will release it at the end of the week :)

**Comment by BorysTyminski** [comment]
Great I really appreciate your hard work. Thank you. Now this component is just perfect. I don't really see other vue compontnt for vue slider which would be better at least at github. Thank you once again.

**Comment by BorysTyminski** [comment]
Are your sure it's working correctly? When just for testing I set delay to 1sec I feel like entire slider is shown only when all images are loaded because I don't need to wait more that 1sec for first slide animation. I use `4.0.0`.

Maybe it have something to do that I'm loading images on custom event like this:

> in parent

    axios.get(`/buildings/getOne.php?id=` + this.articleId, axiosConfig)
        .then(res => {
            this.building = res.data.record[0];
            eventBus.$emit('loadedOffer', this.building);

    <main-offer v-if="building.type !== 'investment'"></main-offer>

> in child which is <main-offer>:

    data () {
        return {
            fluxOptions: {
                autoplay: true,
                delay: 1000,
            },
            fluxImages: [],
            fluxTransitions: {
                transitionSlide2d: Transitions.transitionSlide2d
            }
        };
    },
    created() {
        eventBus.$on('loadedOffer', (data) => {
            this.fluxImages = data.imagesLink.split(',');
            this.$refs.slider.preloadImages(this.fluxImages);
        })
    }

When slider show I don't need to wait more then 1000ms to see next image which is weird because `this.fluxImages` is array of 20 images (this amount just for testing purposes). So I thought that I'll be like 1st image shown as soon as possible then rest images loading "in background" and after few seconds (not one) this slider will start sliding (showing next images)

**Comment by deulos** [comment]
I'll take a look in detail this week but my feeling is that is working properly...

**Comment by deulos** [comment]
Just a clarification, the first image is the first to be requested to server but not the only one, more images are requested to be loaded in parallel. The amount of images loading in parallel is handled by the browser.
